src/solver/constraints-builder/cbuilder.cpp

```diff
@@ -106,124 +103,107 @@ bool CBuilder::update(bool applyCheckBox)
             (*i)->ptr->useLoopFlow = includeLoopFlow;
 
             (*i)->ptr->usePST = includePhaseShift;
-            // onConnectionChanged(link->ptr);
-            // OnStudyLinkChanged(link->ptr);
         }
         // set used to count the number of different impedances
         std::set<double> impedances;
 
         uint columnImpedance = (uint)Data::fhlImpedances;
-        /*
-        //! NTC Direct
-fhlNTCDirect = 0,
-//! NTC Indirect
-fhlNTCIndirect,
-//! Loop flow,
-fhlLoopFlow,
-/! P.shift min,
-fhlPShiftMinus,
-//! P.shift max,
-fhlPShiftPlus,*/
-
-        auto k = *i;
-        // load the impedance
-        auto dataLink = k->dataLink;
 
+        // load the impedance
         // Can probably be improved (below) !!!
-        k->nImpedanceChanges = 0;
-        k->avgImpedance
-          = dataLink->entry[columnImpedance][0]; // or used a function created in matrix.h
+        (*i)->nImpedanceChanges = 0;
+        (*i)->avgImpedance = (*i)->ptr->parameters[columnImpedance][0];
```

Comment:
2 de-reference in a single instruction. Consider using a variable for `*i`.

Reply:
**(*i)** is now replaced with **linkInfo**